lib/dal/src/pkg/import.rs

```diff
@@ -453,125 +453,178 @@ async fn import_schema(
+    // IDs, then the first variant will be considered the default variant in the package. That
+    // sounds completely nuts, but hear me out: in the land before time, packages did not know
+    // their default schema variants and variant specs did not have unique IDs. How did we know
+    // which one was the default back then? We didn't. We just YOLO made the first one the default.
+    // Yeehaw. How do we get around this in a world where they can be mixed and matched? You just
+    // don't trust anything at all whatsoever. Essentially, we will look at all the variants just
+    // to be sure that we have found the default variant or we will fallback to the first.
+    let default_variant_spec_index =
+        match determine_default_variant_spec_index(data, &variant_specs_in_schema_spec) {
+            Some(found_default_variant_spec_index) => found_default_variant_spec_index,
+            None => {
+                // This is only possible if there are no variants in the spec.
+                return Ok(Vec::new());
             }
+        };
 
-            installed_schema_variant_ids.push(variant_id);
+    let mut installed_schema_variant_ids = Vec::new();
+    for (index, variant_spec) in variant_specs_in_schema_spec.iter().enumerate() {
+        let variant = import_schema_variant(
+            ctx,
+            &schema,
+            schema_spec.clone(),
+            variant_spec,
+            installed_module.clone(),
+            thing_map,
+            None,
+        )
+        .await?;
 
-            set_default_schema_variant_id(
-                ctx,
-                &mut schema,
-                schema_spec
-                    .data()
-                    .as_ref()
-                    .and_then(|data| data.default_schema_variant()),
-                variant_spec.unique_id(),
-                variant_id,
-            )
-            .await?;
+        let variant_id = variant.id();
+        installed_schema_variant_ids.push(variant_id);
+
+        // If we are working with the package's default schema variant, then we need to figure out
+        // how to handle it.
+        //
+        // If the schema already existed, then we must leave the package's default variant unlocked
+        // (again, the one we are currently working with). This is because we need to ensure that
+        // the user's local changes to the assets are not automatically clobbered when installing
+        // updated assets from the module index. Not only that, but its corresponding asset func
+        // needs to be unlocked so that the user can regenerate and make changes at will, so we
+        // will do that too.
+        //
+        // If the schema did not already exist, we can safely set the default schema variant and we
+        // don't care about whether or not it becomes locked.
+        let mut can_lock = true;
+        if index == default_variant_spec_index {
```

Comment:
What happens if there is already an unlocked variant in this change set for this schema?

Reply:
This is stopped at the `install_module` route level. Do you think we should be checking that in here instead?